packages/pangea-sdk/pangea/verify_audit.py

```diff
@@ -290,16 +305,32 @@ def _verify_consistency_proof(leaf_index: Optional[int]) -> Status:
 
     else:
         try:
-            curr_root = pub_roots[leaf_index + 1]
-            prev_root = pub_roots[leaf_index]
-            curr_root_hash = decode_hash(curr_root.root_hash)
-            prev_root_hash = decode_hash(prev_root.root_hash)
-            if curr_root.consistency_proof is None:
-                raise ValueError("Consistency proof is missing")
-            logger.debug("Calculating the proof")
-            proof = decode_consistency_proof(curr_root.consistency_proof)
-            if verify_consistency_proof(curr_root_hash, prev_root_hash, proof):
+            if _consistency_proof_ok(pub_roots, leaf_index):
                 status = Status.SUCCEEDED
+
+            elif audit:
+                # Due to an (already fixed) bug, some proofs from Arweave may be wrong.
+                # Try the proof from Pangea instead. If the root hash in both Arweave and Pangea is the same,
+                # it doesn't matter where the proof came from.
+                logger.debug("Consistency proof from Arweave failed to verify")
+                size = leaf_index + 1
+                logger.debug(f"Fetching root from Pangea for size {size}")
+                new_roots = get_pangea_roots(tree_name, [size])
+                if size not in new_roots:
+                    raise ValueError("Error fetching root from Pangea")
+                pangea_roots[size] = new_roots[size]
+                pub_roots[size] = pangea_roots[size]
+                logger.debug(f"Comparing Arweave root hash with Pangea root hash")
+                if pangea_roots[size].root_hash != arweave_roots[size].root_hash:
+                    raise ValueError("Hash does not match")
+                pangea_roots[size] = pangea_roots[size]
+
+                # check again
+                if _consistency_proof_ok(pub_roots, leaf_index):
+                    status = Status.SUCCEEDED
+                else:
+                    status = Status.FAILED
+
             else:
```

Comment:
Here should print the message that need to set PANGEA_TOKEN and PANGEA_DOMAIN or provide them by args

Reply:
done